components/ChargeBar/ChargeBar.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import {
+  components,
+  spacing as themeSpacing,
+  baseFontSize as defaultBaseFontSize,
+} from '../shared/theme';
+import { hexToRgba } from '../shared';
+import Icon from '../Icon';
+
+const Bar = styled.div`
+  border-radius: 4px;
+  overflow: hidden;
+  position: relative;
+
+  ${({ width, background, spacing: { xxsmall } }) => `
+    background-color: ${hexToRgba(background, 0.39)};
+    height: 4px;
+    width: ${width}; 
+    margin-top: ${xxsmall}px;
+  `}
+`;
+
+const Content = styled.div`
+  border-radius: 8px;
+  display: inline-block;
+  height: 100%;
+  left: 0;
+  position: absolute;
+  top: 0;
+  transition: width 0.4s linear;
+
+  ${({ background, progress }) => `
+    background-color: ${background};
+    width: ${progress}%;
+  `}
+`;
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  font-stretch: normal;
+  font-style: normal;
+  font-weight: normal;
+  letter-spacing: 0.18px;
+  line-height: 1.5;
+`;
```

Comment:
Man, here we have some typography properties. How about include width, font-size and color here too?

```suggestion
const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0.18px;
  line-height: 1.5;
  
  ${({ width, background, baseFontSize }) => `
    width: ${width};
    color: ${background};
    font-size: ${baseFontSize * 0.875}px;
  `}
`;
```